libpod/runtime_ctr.go

```diff
@@ -47,6 +47,32 @@ func (r *Runtime) NewContainer(ctx context.Context, rSpec *spec.Spec, options ..
 	return r.newContainer(ctx, rSpec, options...)
 }
 
+func (r *Runtime) PrepareVolumeOnCreateContainer(ctx context.Context, ctr *Container) (retErr error) {
```

Comment:
Why did you move to a separate function for this, as opposed to keeping it part of container creation?

Reply:
When I had this code originally in `setupContainer()`, then `podman create` would hang indefinitely. Only a process kill would make the operation abort. I guess some of the deferred calls in the `setupContainer()` need to be executed before calling the volume copy.

So, I made the change to look it somewhat like `podman create` followed by `podman init` so as to follow the same business logic (code path).